backend/src/main/java/ch/puzzle/pcts/service/business/ExperienceCalculationBusinessService.java

```diff
@@ -0,0 +1,136 @@
+            throw new PCTSException(HttpStatus.NOT_FOUND, List.of(error));
+        }
+        experienceCalculationValidationService.validateOnUpdate(id, experienceCalculation);
+        experienceCalculationValidationService.validateMemberForCalculation(experienceCalculation);
+        experienceCalculation.setId(id);
+        return experienceCalculationPersistenceService.save(experienceCalculation);
+    }
+
+    public List<ExperienceCalculation> createExperienceCalculations(Calculation calculation) {
+        return calculation.getExperiences().stream().map(exp -> {
+            exp.setCalculation(calculation);
+            return this.create(exp);
+        }).toList();
+    }
+
+    public List<ExperienceCalculation> updateExperienceCalculations(Calculation calculation) {
+        List<ExperienceCalculation> existing = this.getByCalculationId(calculation.getId());
+
+        List<ExperienceCalculation> experienceCalculations = calculation.getExperiences().stream().map(exp -> {
+            exp.setCalculation(calculation);
+
+            Optional<Long> expCalcId = findIdByCalculationAndExperience(exp, existing);
+
+            return expCalcId.isEmpty() ? this.create(exp) : this.update(expCalcId.get(), exp);
+        }).toList();
+
+        /*
+         * Removing all created or updated experience calculations to later delete the
+         * unused experience calculations
+         */
+        existing.removeAll(experienceCalculations);
+        existing.stream().map(ExperienceCalculation::getId).forEach(this::delete);
+
+        return experienceCalculations;
+    }
+
+    private Optional<Long> findIdByCalculationAndExperience(ExperienceCalculation experienceCalculation,
+                                                            List<ExperienceCalculation> experienceCalculationList) {
+        return experienceCalculationList
+                .stream()
+                .filter(ec -> ec.getCalculation().getId().equals(experienceCalculation.getCalculation().getId())
+                              && ec.getExperience().getId().equals(experienceCalculation.getExperience().getId()))
+                .map(ExperienceCalculation::getId)
+                .findFirst();
+    }
+
+    /*
+     * The divisions are rounded to a DECIMAL128 digit because numbers with a
+     * infinite amount of digits could cause a ArithmeticException
+     */
+    private BigDecimal calculatePoints(ExperienceCalculation calculation) {
+        Experience experience = calculation.getExperience();
+        ExperienceType type = experience.getType();
+        Relevancy relevancy = calculation.getRelevancy();
+
+        BigDecimal basePoints = type.getPoints(relevancy);
+
+        BigDecimal percentFactor = BigDecimal
+                .valueOf(experience.getPercent())
+                .divide(BigDecimal.valueOf(100), MathContext.DECIMAL128);
+
+        long days = experience.getStartDate().until(experience.getEndDate(), ChronoUnit.DAYS);
```

Comment:
The method LocalDate.until(end, ChronoUnit.DAYS) calculates the difference excluding the end date.
Are you aware of that?